asset-transfer-basic/application-typescript-hsm/README.md

```diff
@@ -0,0 +1,240 @@
+and a user pin of `98765432`, however you can override these values to use your own HSM by either
+editting the application or use these environment variables to pass in the values:
+
+* PKCS11_LIB - path to the your specific HSM Library
+* PKCS11_PIN - your HSM pin
+* PKCS11_LABEL - your HSM label
+
+Alternatively you could install SoftHSM to try out the application as described in the next sections
+
+### Install SoftHSM
+
+In order to run the application in the absence of a real HSM, a software
+emulator of the PKCS#11 interface is required.
+For more information please refer to [SoftHSM](https://www.opendnssec.org/softhsm/).
+
+SoftHSM can either be installed using the package manager for your host system:
+
+* Ubuntu: `sudo apt install softhsm2`
+* macOS: `brew install softhsm`
+* Windows: **unsupported**
+
+Or compiled and installed from source:
+
+1. install openssl 1.0.0+ or botan 1.10.0+
+2. download the source code from <https://dist.opendnssec.org/source/softhsm-2.5.0.tar.gz>
+3. `tar -xvf softhsm-2.5.0.tar.gz`
+4. `cd softhsm-2.5.0`
+5. `./configure --disable-gost` (would require additional libraries, turn it off unless you need 'gost' algorithm support for the Russian market)
+6. `make`
+7. `sudo make install`
+
+### Specify the SoftHSM configuration file
+A configuration file for SoftHSM is provided in this sample directory. This file
+uses /tmp as the location for SoftHSM to store it's data which means (depending on
+your operating system configuration) the data could be deleted at some point, for example
+when you reboot your system. If this data is lost then you will have to delete the wallet
+created. An alternative is to change this file to store SoftHSM data in a permanent location
+on your file system.
+To use this configuration file you need to export an environment variable to point to it
+for example, if you are in the application directory then you can use the following command
+
+```bash
+export SOFTHSM2_CONF=$PWD/softhsm2.conf
+```
+
+Ensure you have this set when initializing the token as well as running the application
+
+### Initialize a token to store keys in SoftHSM
+If you have not initialized a token previously (or it has been deleted) then you will need to perform this one time operation
+
+```bash
+softhsm2-util --init-token --slot 0 --label "ForFabric" --pin 98765432 --so-pin 1234
+```
+
+The Security Officer PIN, specified with the `--so-pin` flag, can be used to re-initialize the token,
+and the user PIN (see below), specified with the `--pin` flag, is used by applications to access the token for
+generating and retrieving keys.
+
+
+## Running the sample
+
+Use the Fabric test network utility (`network.sh`) to start a Fabric network and deploy one
+one of the sample chaincodes.
+
+Follow these step in order.
+- Start the test network with a Certificate Authority and create a channel, ensuring you are in the `fabric-samples` directory
+```
+cd test-network
+./network.sh up createChannel -c mychannel -ca
```

Comment:
experience tells me that I need to run `./network.sh down` first... as usually I've got something left over from the last run and nothing works.